xml/src/test/java/org/ehcache/xml/multi/XmlMultiConfigurationTest.java

```diff
@@ -0,0 +1,506 @@
+
+    assertThat(xmlMultiConfiguration.configuration("foo"), sameInstance(config));
+    assertThat(xmlMultiConfiguration.configuration("foo", "prod"), sameInstance(config));
+
+    assertThat(xmlMultiConfiguration.configurations().keySet(), containsInAnyOrder("foo"));
+    assertThat(xmlMultiConfiguration.configurations("prod").keySet(), containsInAnyOrder("foo"));
+
+    assertThat(xmlMultiConfiguration.identities(), containsInAnyOrder("foo"));
+    try {
+      xmlMultiConfiguration.variants();
+      fail("Expected IllegalStateException");
+    } catch (IllegalStateException e) {
+      //expected
+    }
+    try {
+      xmlMultiConfiguration.variants("foo");
+      fail("Expected IllegalStateException");
+    } catch (IllegalStateException e) {
+      //expected
+    }
+
+    assertThat(xmlMultiConfiguration.toString(),
+      isSimilarTo(
+        "<configurations xmlns='http://www.ehcache.org/v3/multi'>" +
+          "<configuration identity='foo'><config xmlns='http://www.ehcache.org/v3'/></configuration>" +
+        "</configurations>").ignoreWhitespace().ignoreComments());
+  }
+
+  @Test
+  public void testVariantConfigurationBuiltFromEmpty() {
+    Configuration barVariant = emptyConfiguration();
+    Configuration bazVariant = emptyConfiguration();
+    XmlMultiConfiguration xmlMultiConfiguration = XmlMultiConfiguration.fromNothing()
+      .withManager("foo").variant("bar", barVariant).variant("baz", bazVariant).build();
+
+    try {
+      xmlMultiConfiguration.configuration("foo");
+      fail("Expected IllegalStateException");
+    } catch (IllegalStateException e) {
+      //expected
+    }
+
+    try {
+      xmlMultiConfiguration.configurations();
+      fail("Expected IllegalStateException");
+    } catch (IllegalStateException e) {
+      //expected
+    }
+
+    assertThat(xmlMultiConfiguration.configuration("foo", "bar"), sameInstance(barVariant));
+    assertThat(xmlMultiConfiguration.configuration("foo", "baz"), sameInstance(bazVariant));
+
+    assertThat(xmlMultiConfiguration.configurations("bar").keySet(), containsInAnyOrder("foo"));
+    assertThat(xmlMultiConfiguration.configurations("baz").keySet(), containsInAnyOrder("foo"));
+
+    assertThat(xmlMultiConfiguration.identities(), containsInAnyOrder("foo"));
+    assertThat(xmlMultiConfiguration.variants(), containsInAnyOrder("bar", "baz"));
+    assertThat(xmlMultiConfiguration.variants("foo"), containsInAnyOrder("bar", "baz"));
+
+    assertThat(xmlMultiConfiguration.toString(), isSimilarTo(
+      "<configurations xmlns='http://www.ehcache.org/v3/multi'>" +
+        "<configuration identity='foo'>" +
+          "<variant type='bar'><config xmlns='http://www.ehcache.org/v3'/></variant>" +
+          "<variant type='baz'><config xmlns='http://www.ehcache.org/v3'/></variant>" +
+        "</configuration>" +
+      "</configurations>").ignoreWhitespace().ignoreComments());
+  }
+
+  @Test
+  public void testMixedConfigurationBuiltFromEmpty() {
```

Comment:
This test is missing a variant scenario -- how about adding `.withManager("fum").variant("bar", barVariant)` and see what happens?  One thing this points out is that `XmlMultiConfiguration.configurations(variant)` and `XmlMultiConfiguration.variants(identity)` throw different exceptions -- maybe they should be the same.